pkgs/development/compilers/rust/binary.nix

```diff
@@ -46,6 +46,9 @@ rec {
         patchelf \
           --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker) \
           "$out/bin/rustc"
+        # rustc bootstrap needs libz
```

Comment:
Not sure if it's really necessary, since this will go to staging anyways, but it wouldn't hurt to make sure this is only added for the new versions that need zlib, so that the old versions stay the same and can be cached.

Reply:
added a `stdenv.lib.versionAtLeast` for that

Reply:
@Kloenk I think @petabyteboy means making that itself comment be outside all string literals so that it doesn't force rebuilds.

That sounds to me like general good practice that I would agree with.